### PR TITLE
2021.2

### LINKED ISSUE #555: Update package astropy to 4.2
Update core package astropy to version 4.2.

We currently have 4.0.1post.

### LINKED ISSUE #568: Add package sot/jobwatch 0.2.0
Add package jobwatch 0.2.0

### LINKED ISSUE #583: Add package iniconfig to ska3-core
This is installed as of skare3 2021.1 but is not called out explicitly in what should be the controlling metapackage for this (ska3-core).

### LINKED ISSUE #585: Update sot/perigee_health_plots to Release 0.1.3
Fix the scale in the dac vs dtemp plot.

Includes the following merges:
- PR [#18](https://github.com/sot/perigee_health_plots/pull/18): Use config DAVCSDTEMP_PLOT xlim for dtemp default lims

### LINKED ISSUE #587: Add package Plotly
Please add Plotly to the core collection of Ska packages. Additionally, please add Kaleido to enable the output of Plotly charts to static documents (e.g. png, svg, pdf, etc.). Both packages can be installed via the following commands:

conda install plotly
conda install -c plotly python-kaleido

### LINKED ISSUE #588: Update package bleach to 3.2.3 (or latest)
Update package `bleach` to 3.2.3

This includes a fix for an annoying deprecation warning that shows up in `pytest` in packages like `sparkles` (see below). I confirmed that installing 3.2.3 makes this warning go away.

```
======================================================================== warnings summary ========================================================================
sparkles/tests/test_review.py::test_review_catalog
  /Users/aldcroft/miniconda3/envs/ska3-agasc/lib/python3.8/site-packages/bleach/_vendor/html5lib/_trie/_base.py:3: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    from collections import Mapping

-- Docs: https://docs.pytest.org/en/stable/warnings.html
================================================================= 33 passed, 1 warning in 35.55s =================================================================
```

### LINKED ISSUE #589: Update sot/mica to Release 4.25.0
Includes the following merges:
- PR [#257](https://github.com/sot/mica/pull/257): Avoid vv obspar race condition
- PR [#256](https://github.com/sot/mica/pull/256): Skip report vv warn if no sybase passwd file for db
- PR [#255](https://github.com/sot/mica/pull/255): Fix script default to actually save plots

### LINKED ISSUE #590: Release ska3-flight 2021.2
# Packages to include

- astropy 4.2
- maude
- mica
- plotly

# Checklist

- [x] Check packages are built and copied to test channel
- [x] Document all changes
- [x] Document test status
- [x] Get approval
- [x] Promote packages to main conda channel
- [x] Update production systems:
   - [x] announce to aca@cfa
   - [x] update HEAD ska3/flight as aca user
   - [x] run ska_testr on HEAD
   - [ ] ~update ska3/flight on cheru as sot user~
   - [x] update ska3/flight on chimchim as SOT user
   - [x] announce to chandracode and mpweekly

### LINKED ISSUE #591: Update sot/acdc to Release 4.7.2
Includes the following merges:
- PR [#60](https://github.com/sot/acdc/pull/60): Update report link

### LINKED ISSUE #592: Update sot/chandra_aca to Release 4.32.0
Includes the following merges:
- PR [#114](https://github.com/sot/chandra_aca/pull/114): Add calc_target_offsets function
- PR [#113](https://github.com/sot/chandra_aca/pull/113): Fetch ACA telemetry from MAUDE using blobs

### LINKED ISSUE #593: Update sot/aca_weekly_report to Release 0.1.5
Includes the following merges:
- PR [#9](https://github.com/sot/aca_weekly_report/pull/9): Update task_schedule.cfg

### LINKED ISSUE #594: Update sot/maude to Release 3.6.0
Includes the following merges:
- PR [#33](https://github.com/sot/maude/pull/33): Fix chunks query when MAUDE returns no data in last chunk
- PR [#32](https://github.com/sot/maude/pull/32): Do not assign defaults for start/stop times + exception handling

### LINKED ISSUE #595: Update sot/aca_view to Release 0.6.0
Includes the following merges:
- PR [#65](https://github.com/sot/aca_view/pull/65): Fix so aca_view can realistically be used in real-time mode.

### LINKED ISSUE #596: Update sot/skare3_tools to Release 1.0.3
Includes the following merges:
- PR [#78](https://github.com/sot/skare3_tools/pull/78): Fix release description so PR links are full URIs

### LINKED ISSUE #597: Update sot/kadi to Release 5.5.1
Includes the following merges:
- PR [#199](https://github.com/sot/kadi/pull/199): Fix bug getting states starting between AOUPTARQ and AOMANUVR

### LINKED ISSUE #601: Update sot/cxotime to Release 3.2.4
Includes the following merges:
- PR [#24](https://github.com/sot/cxotime/pull/24): Astropy 4.2 compatibility: modified "import erfa" statement

### LINKED ISSUE #602: Update sot/proseco to Release 4.11.1
Includes the following merges:
- PR [#350](https://github.com/sot/proseco/pull/350): Fix problem from table API change in astropy 4.2

### LINKED ISSUE #605: Update sot/acdc to Release 4.7.3
Includes the following merges:
- PR [#61](https://github.com/sot/acdc/pull/61): Adding a couple of logging messages

### LINKED ISSUE #606: Update sot/cxotime to Release 3.2.5
Includes the following merges:
- PR [#25](https://github.com/sot/cxotime/pull/25): Ignore ERFA dubious year warning

### LINKED ISSUE #607: Update sot/eng_archive to Release 4.52.0
Includes the following merges:
- PR [#214](https://github.com/sot/eng_archive/pull/214): Retry tables open file
- PR [#215](https://github.com/sot/eng_archive/pull/215): Fix sync index column order to match order from astropy 4.0
- PR [#217](https://github.com/sot/eng_archive/pull/217): Fix file_defs.py for PEP8
- PR [#213](https://github.com/sot/eng_archive/pull/213): Update flake8.yml to Python 3.8

### LINKED ISSUE #608: Update sot/ska_helpers to Release 0.5.0
Includes the following merges:
- PR [#21](https://github.com/sot/ska_helpers/pull/21): Add tables_open_file func to retry opening

### PR DESCRIPTION
**Pull request by javierggt**
# ska3-flight 2021.2

## Summary:

This is an update of ska3-core and ska3-flight. The most notable change is the update of astropy to version 4.2. A test installation is available for testing at `/proj/sot/ska3/test-2021.2rc5` on HEAD and GRETA linux, as announced on February 19 2021 on the chandracode distribution list.

Fixes/changes to maude, kadi and chandra_aca will be briefed by Tom (shown in bold in the Code Changes section below).

This release also includes some minor fixes and changes to several packages. Two packages that can be of general use were added to ska3-core: mamba and plotly.

### astropy 4.2

astropy 4.2 introduces some API changes that ended up impacting our code in regression testing (https://docs.astropy.org/en/latest/changelog.html#id91):

- Change Table.columns.keys() and Table.columns.values() to both return generators instead of a list. This matches the behavior for Python dict objects. [#10543]
- The order of columns when creating a table from a list of dict may be changed. Previously, the order was alphabetical because the dict keys were assumed to be in random order. Since Python 3.7, the keys are always in order of insertion, so Table now uses the order of keys in the first row to set the column order. To alphabetize the columns to match the previous behavior, use t = t[sorted(t.colnames)]. [#10900]

## Testing:

Automated regression testing: https://icxc.cfa.harvard.edu/aspect/skare3/testr/releases/2021.2rc5/

The one apparent failure is actually a warning due to insufficient access permissions. This test does pass if it is run with the right permissions.

Testing in `/proj/sot/ska3/test-2021.2rc5`:
- [kady](https://icxc.cfa.harvard.edu/aspect/skare3/testr/releases/2021.2rc5-kady)
- [chimchim](https://icxc.cfa.harvard.edu/aspect/skare3/testr/releases/2021.2rc5-chimchim)

## Review

All operations critical or impacting PR's are independently and carefully reviewed. For other PR's the level of detail for review is calibrated to operations criticality. Some PR's that are confined to aspect-team-specific processing may have little to no independent review.

## Deployment:
-  The 2021.2 release will be promoted to the flight conda channel, and deployed on HEAD and GRETA ska3/flight after the release and products for the week are approved.

# Code changes

### New Packages
- **jobwatch: 0.2.0**

### Updated Packages

- **ska3-core:** 2021.1 -> 2021.2

- **aca_view:** 0.5.0 -> 0.6.0 (0.5.0 -> 0.6.0)
  - [PR 65](https://github.com/sot/aca_view/pull/65) (javierggt): Fix so aca_view can realistically be used in real-time mode.
- **aca_weekly_report:** 0.1.4 -> 0.1.5 (0.1.4 -> 0.1.5)
  - [PR 9](https://github.com/sot/aca_weekly_report/pull/9) (jeanconn): Update task_schedule.cfg
- **acdc:** 4.7.1 -> 4.7.3 (4.7.1 -> 4.7.2 -> 4.7.3)
  - [PR 60](https://github.com/sot/acdc/pull/60) (jeanconn): Update report link
  - [PR 61](https://github.com/sot/acdc/pull/61) (jeanconn): Adding a couple of logging messages
- **chandra_aca:** 4.31.1 -> 4.32.0 (4.31.1 -> 4.32.0)
  - [PR 113](https://github.com/sot/chandra_aca/pull/113) (javierggt): Fetch ACA telemetry from MAUDE using blobs
  - **[PR 114](https://github.com/sot/chandra_aca/pull/114) (taldcroft): Add calc_target_offsets function**
- **cxotime:** 3.2.3 -> 3.2.5 (3.2.3 ->  3.2.4 -> 3.2.5)
  - [PR 24](https://github.com/sot/cxotime/pull/24) (taldcroft): Astropy 4.2 compatibility: modified "import erfa" statement
  - [PR 25](https://github.com/sot/cxotime/pull/25) (javierggt): Ignore ERFA dubious year warning
- **kadi:** 5.5.0 -> 5.5.1 (5.5.0 -> 5.5.1)
  - **[PR 199](https://github.com/sot/kadi/pull/199) (taldcroft): Fix bug getting states starting between AOUPTARQ and AOMANUVR**
- **maude:** 3.5.0 -> 3.6.0 (3.5.0 -> 3.6.0)
  - **[PR 32](https://github.com/sot/maude/pull/32) (taldcroft): Do not assign defaults for start/stop times + exception handling**
  - [PR 33](https://github.com/sot/maude/pull/33) (taldcroft): Fix chunks query when MAUDE returns no data in last chunk
- **mica:** 4.24.0 -> 4.25.0 (4.24.0 -> 4.25.0)
  - [PR 255](https://github.com/sot/mica/pull/255) (jeanconn): Fix script default to actually save plots
  - [PR 256](https://github.com/sot/mica/pull/256) (jeanconn): Skip report vv warn if no sybase passwd file for db
  - [PR 257](https://github.com/sot/mica/pull/257) (jeanconn): Avoid vv obspar race condition
- **perigee_health_plots:** 0.1.2 -> 0.1.3 (0.1.2 -> 0.1.3)
  - [PR 18](https://github.com/sot/perigee_health_plots/pull/18) (jeanconn): Use config DAVCSDTEMP_PLOT xlim for dtemp default lims
- **proseco:** 4.11.0 -> 4.11.1 (4.11.0 -> 4.11.1)
  -  [PR 350](https://github.com/sot/proseco/pull/350) (taldcroft): Fix problem from table API change in astropy 4.2
- **ska.engarchive:** 4.51.0 -> 4.52.0 (4.51.0 -> 4.52.0)
  - [PR 213](https://github.com/sot/eng_archive/pull/213) (taldcroft): Update flake8.yml to Python 3.8
  - [PR 217](https://github.com/sot/eng_archive/pull/217) (taldcroft): Fix file_defs.py for PEP8
  - [PR 215](https://github.com/sot/eng_archive/pull/215) (jeanconn): Fix sync index column order to match order from astropy 4.0
  - [PR 214](https://github.com/sot/eng_archive/pull/214) (taldcroft): Retry tables open file
- **ska_helpers:** 0.4.0 -> 0.5.0 (0.4.0 -> 0.5.0)
  - [PR 21](https://github.com/sot/ska_helpers/pull/21) (taldcroft): Add tables_open_file func to retry opening
- **skare3_tools:** 1.0.2 -> 1.0.3 (1.0.2 -> 1.0.3)
  - [PR 78](https://github.com/sot/skare3_tools/pull/78) (javierggt): Fix release description so PR links are full URIs

### New Packages
- **iniconfig: 1.1.1**
- **krb5: 1.18.2**
- **libcurl: 7.71.1**
- **libsolv: 0.7.14**
- **libssh2: 1.9.0**
- **mamba: 0.4.2**
- **plotly: 4.14.3**
- **pyerfa: 1.7.1.1**
- **python-kaleido: 0.0.3**
- **python_abi: 3.8**
- **retrying: 1.3.3**

### Updated Packages

- **astropy:** 4.0.1.post1 -> 4.2
- **bleach:** 3.1.5 -> 3.3.0
- **ca-certificates:** 2020.6.24 -> 2021.1.19
- **certifi:** 2020.6.20 -> 2020.12.5
- **openssl:** 1.1.1g -> 1.1.1i
- **pytest:** 6.2 -> 6.2.0

# Related Issues

Fixes #555
Fixes #568
Fixes #583
Fixes #585
Fixes #587
Fixes #588
Fixes #589
Fixes #590
Fixes #591
Fixes #592
Fixes #593
Fixes #594
Fixes #595
Fixes #596
Fixes #597
Fixes #601
Fixes #602
Fixes #605
Fixes #606
Fixes #607
Fixes #608

### CODE REVIEW

File: pkg_defs/ska3-core-latest/meta.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ requirements:
    - jupyterlab
    - line_profiler
    - lxml
+   - mamba
    - matplotlib
```
Comment (taldcroft) [inline pkg_defs/ska3-core-latest/meta.yaml:30]: Probably also `pyplot`, since that is not a dependency of anything else.

**Review by taldcroft — COMMENTED** [review]
I skimmed through this and it looks reasonable. It would be good for @jeanconn to use her eye for detail to make sure all the package numbers are right.

File: pkg_defs/ska3-core-latest/meta.yaml (hunk 1 above)
Comment (javierggt) [reply pkg_defs/ska3-core-latest/meta.yaml:30]: pyplot is not in ska3-core

Comment (jeanconn) [reply pkg_defs/ska3-core-latest/meta.yaml:30]: Or if Tom meant plotly, he already got that in in a separate PR.

**Comment by matthewdahmer** [comment]
I didn't see Kaleido listed in the pkg_defs/ska3-core/meta.yaml file, though I did see Plotly (thank you). Kaleido is required to export Plotly-generated images to a file from a Jupyter notebook and was listed in the issue/request. If it is included in ska3-core via some other means, please disregard this message.

**Comment by javierggt** [comment]
> I didn't see Kaleido listed in the pkg_defs/ska3-core/meta.yaml file..

Thanks for reminding me, and I am sorry I missed it. I just uploaded the package to the conda channel and modified meta.yml, so it should be available in 2021.2rc4.

**Comment by matthewdahmer** [comment]
Thank You!

**Pull request merged**